### PR TITLE
Add crkbd/gotham keymap

### PR DESCRIPTION
**Pull request by GauthamYerroju**
Add my custom keymap for Corne.

## Description

Adds a keymap for Corne keyboard. The keymap is meant to be an easy transition into 40% layouts. Numbers are in the top row, combined with navigation cluster. The = symbol has a dedicated key because it's so common in programming. Most used symbols are in easily reachable positions.

Also adds custom OLED font and code.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/keymaps/gotham/config.h [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+#define NO_ACTION_ONESHOT
+
+#ifdef AUDIO_ENABLE
+    #define B5_AUDIO
+    #define NO_MUSIC_MODE
+    #define AUDIO_CLICKY
+#endif
+
+#define OLED_FONT_H "keyboards/crkbd/keymaps/gautham/glcdfont.c"
```
Comment (drashna) [inline keyboards/crkbd/keymaps/gotham/config.h:45]:
```suggestion
#define OLED_FONT_H "keyboards/crkbd/keymaps/gotham/glcdfont.c"
```

File: keyboards/crkbd/keymaps/gotham/keycodes.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 3
```
Comment (drashna) [inline keyboards/crkbd/keymaps/gotham/keycodes.h:6]: Would prefer to see this as an enum, as it's easier to maintain, long term. 

```suggestion
enum layers {
    _QWERTY,
    _LOWER,
    _RAISE,
    _ADJUST,
};
```

File: keyboards/crkbd/keymaps/gotham/keymap.c [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  if (state == default_layer_state) {
+    restore_rgb_config();
+  } else {
+    uint8_t layer = biton32(state);
+    if (layer_state == default_layer_state) save_rgb_config();
+    rgb_by_layer(layer);
+  }
+  return state;
+}
+
```
Comment (drashna) [inline keyboards/crkbd/keymaps/gotham/keymap.c:75]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
  if (state == default_layer_state) {
    restore_rgb_config();
  } else {
    uint8_t layer = get_highest_layer(state);
    if (layer_state == default_layer_state) save_rgb_config();
    rgb_by_layer(layer);
  }
  return state;
}

```

File: keyboards/crkbd/keymaps/gotham/rgb.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+
+void save_rgb_config(void) {
+  #ifdef RGBLIGHT_ENABLE
+    RGB_current_config.enable = rgblight_config.enable;
+    RGB_current_config.mode = rgblight_config.mode;
+    RGB_current_config.speed  = rgblight_config.speed;
+    RGB_current_config.hue = rgblight_config.hue;
+    RGB_current_config.sat = rgblight_config.sat;
+    RGB_current_config.val = rgblight_config.val;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/gotham/rgb.c:21]: Most of these can be retrieved by a function:
https://github.com/qmk/qmk_firmware/blob/master/quantum/rgblight.h#L297-L305

**Comment by GauthamYerroju** [comment]
Apologies, I'm refactoring my workflow and I messed up. Deleting merge request for now.

**Pull request closed without merging**